src/siminterface/simulator.py

```diff
@@ -234,3 +242,19 @@ def network_metrics(self):
     def get_active_ingress_nodes(self):
         """Return names of all ingress nodes that are currently active, ie, produce flows."""
         return [ing[0] for ing in self.ing_nodes if self.params.inter_arr_mean[ing[0]] is not None]
+
+
+# for debugging
+if __name__ == "__main__":
+    # run from project root for file paths to work
+    # I changed triangle to have 2 ingress nodes for debugging
+    network_file = 'params/networks/triangle.graphml'
+    service_file = 'params/services/abc.yaml'
+    config_file = 'params/config/sim_config.yaml'
+
+    sim = Simulator(network_file, service_file, config_file)
+    state = sim.init(seed=1234)
+    dummy_action = SimulatorAction(placement={}, scheduling={})
+    # FIXME: this currently breaks - negative flow counter?
```

Comment:
Did you test this?

Reply:
I left this for another issue, I will patch in a separate PR